sharing_portal/views.py

```diff
@@ -1,226 +1,186 @@
-import json
-
-from csp.decorators import csp_update
 from django.conf import settings
 from django.contrib import messages
 from django.contrib.auth.decorators import login_required
 from django.core.mail import send_mail
 from django.urls import reverse
-from django.db.models import F, Q, IntegerField, Count
 from django.http import Http404, HttpResponse, HttpResponseRedirect
 from django.core.exceptions import PermissionDenied
-from django.shortcuts import get_object_or_404, redirect
+from django.shortcuts import redirect
 from django.template import loader
 from django.utils.html import strip_tags
 from django.utils import timezone
 from projects.models import Project
-from allocations.models import Allocation
-from rest_framework.renderers import JSONRenderer
-from rest_framework import serializers
+from projects.util import get_project_members
 from util.project_allocation_mapper import ProjectAllocationMapper
 from util.keycloak_client import KeycloakClient
 
 from .forms import (
     ArtifactForm,
-    ArtifactVersionForm,
     AuthorFormset,
     ShareArtifactForm,
-    ZenodoPublishFormset,
     RequestDaypassForm,
     ReviewDaypassForm,
 )
-from .models import Artifact, ArtifactVersion, Author, ShareTarget, DaypassRequest
-from .tasks import publish_to_zenodo
+from .models import DaypassRequest, DaypassProject
+from . import trovi
 from projects.views import (
     add_project_invitation,
     get_invite_url,
     get_project_membership_managers,
     is_membership_manager,
     manage_membership_in_scope,
 )
+from .zenodo import ZenodoClient
+
+from urllib.parse import urlencode
 
 import logging
+from datetime import datetime, timedelta
 
 LOG = logging.getLogger(__name__)
 
 SHARING_KEY_PARAM = "s"
 
 
-class ArtifactSerializer(serializers.ModelSerializer):
-    class Meta:
-        model = Artifact
-        fields = "__all__"
+def with_trovi_token(view_func):
+    def _wrapped_view(request, *args, **kwargs):
+        if (
+            request.session.get("trovi_token_expiration")
+            and datetime.utcnow().timestamp() > request.session["trovi_token_expiration"]
```

Comment:
I mentioned in a comment yesterday, but it might be good to give yourself some buffer room in case of weird timing causing the token to expire before trovi validates it.